scylla/tests/integration/statement.rs

```diff
@@ -309,3 +317,754 @@ async fn test_timestamp_generator() {
         .map(|row_result| row_result.unwrap())
         .all(|(_a, _b, writetime)| timestamps_locked.contains(&writetime)));
 }
+
+#[tokio::test]
+async fn test_named_bind_markers() {
+    let session = create_new_session_builder().build().await.unwrap();
+    let ks = unique_keyspace_name();
+
+    session
+        .ddl(format!("CREATE KEYSPACE {} WITH REPLICATION = {{'class' : 'NetworkTopologyStrategy', 'replication_factor' : 1}}", ks))
+        .await
+        .unwrap();
+
+    session
+        .query_unpaged(format!("USE {}", ks), &[])
+        .await
+        .unwrap();
+
+    session
+        .ddl("CREATE TABLE t (pk int, ck int, v int, PRIMARY KEY (pk, ck, v))")
+        .await
+        .unwrap();
+
+    session.await_schema_agreement().await.unwrap();
+
+    let prepared = session
+        .prepare("INSERT INTO t (pk, ck, v) VALUES (:pk, :ck, :v)")
+        .await
+        .unwrap();
+    let hashmap: HashMap<&str, i32> = HashMap::from([("pk", 7), ("v", 42), ("ck", 13)]);
+    session.execute_unpaged(&prepared, &hashmap).await.unwrap();
+
+    let btreemap: BTreeMap<&str, i32> = BTreeMap::from([("ck", 113), ("v", 142), ("pk", 17)]);
+    session.execute_unpaged(&prepared, &btreemap).await.unwrap();
+
+    let rows: Vec<(i32, i32, i32)> = session
+        .query_unpaged("SELECT pk, ck, v FROM t", &[])
+        .await
+        .unwrap()
+        .into_rows_result()
+        .unwrap()
+        .rows::<(i32, i32, i32)>()
+        .unwrap()
+        .map(|res| res.unwrap())
+        .collect();
+
+    assert_eq!(rows, vec![(7, 13, 42), (17, 113, 142)]);
+
+    let wrongmaps: Vec<HashMap<&str, i32>> = vec![
+        HashMap::from([("pk", 7), ("fefe", 42), ("ck", 13)]),
+        HashMap::from([("xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx", 7)]),
+        HashMap::new(),
+        HashMap::from([("ck", 9)]),
+    ];
+    for wrongmap in wrongmaps {
+        assert!(session.execute_unpaged(&prepared, &wrongmap).await.is_err());
+    }
```

Comment:
Instead of only checking that it results in an error, could you also verify what this error is?
In other tests we did have situations that unrelated bug (e.g. wrong query string) throws an error, resulting in a test being useless.